test/workers/hermes_worker.js

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright: Ambrosus Technologies GmbH
+Email: tech@ambrosus.com
+
+This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+This Source Code Form is “Incompatible With Secondary Licenses”, as defined by the Mozilla Public License, v. 2.0.
+*/
+
+import chai from 'chai';
+import sinon from 'sinon';
+import HermesWorker from '../../src/workers/hermes_worker';
+
+const {expect} = chai;
+
+describe('Hermes Worker', () => {
+  const bundleSizeLimit = 10;
+  const storagePeriods = 1;
+  let mockDataModelEngine;
+  let mockConfig;
+  let mockLogger;
+  let mockResult;
+  let hermesWorker;
+
+  beforeEach(async () => {
+    mockDataModelEngine = {
+      finaliseBundle: sinon.stub()
+    };
+    mockConfig = {
+      bundleSizeLimit,
+      storagePeriods
+    };
+    mockLogger = {
+      info: sinon.stub()
+    };
+    mockResult = {
+      bundleId: '0xc0ffee'
+    };
+    hermesWorker = new HermesWorker(mockDataModelEngine, mockConfig, mockLogger);
```

Comment:
I'm missing a call to `beforeWorkLoop` here, which means the worker may not be fully initialised.